src/txn/txn_timestamp.c

```diff
@@ -1056,98 +1122,207 @@ __wt_txn_publish_commit_timestamp(WT_SESSION_IMPL *session)
 	 * fixed.
 	 */
 	ts = txn->commit_timestamp;
+	txn->first_commit_timestamp = ts;
 
-	__wt_writelock(session, &txn_global->commit_timestamp_rwlock);
+	/*
+	 * Provided that we don't already have an explicit durable timestamp set
+	 * (indicated by WT_TXN_HAS_TS_DURABLE) and we haven't published a
+	 * durable timestamp yet (indicated by WT_TXN_PUBLIC_TS_DURABLE), let's
+	 * insert the txn in the durable queue using its commit timestamp as an
+	 * implied durable timestamp.
```

Comment:
This comment doesn't seem to match the code - since the code isn't checking for `WT_TXN_PUBLIC_TS_DURABLE`. It's also generally preferable to avoid referencing variable names in comments - they should be standalone description of the functionality not a restatement of the code logic.

Reply:
3e0eab0

This part was refactored and the comment itself was removed as it was no longer relevant.